django/db/backends/postgresql/client.py

```diff
@@ -54,8 +55,11 @@ def runshell_db(cls, conn_params):
                     # If the current locale can't encode the data, we let
                     # the user input the password manually.
                     pass
+            default_handler = signal.getsignal(signal.SIGINT)
+            signal.signal(signal.SIGINT, lambda sig, frame: None)
             subprocess.check_call(args)
         finally:
+            signal.signal(signal.SIGINT, default_handler)
```

Comment:
What about just moving the `default_handler` assignment up above the try block? I've pushed a commit with that change.